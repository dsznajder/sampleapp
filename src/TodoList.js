import React, { Component } from 'react';

export default class TodoList extends Component {
  completeTodo = (todo) => () => this.props.completeTodo(todo);
  removeTodo = (todo) => () => this.props.removeTodo(todo);

  render() {
    return (
      <ul>
        {this.props.todos.map((todo, index) => (
          <li key={index}>
            <div>
              <a id="todo" className={todo.completed ? 'done text-danger' : 'text-primary'} onClick={this.completeTodo(todo)}>
                {todo.text}
              </a>

              <a id="remove" className="btn btn-danger" onClick={this.removeTodo(todo)}> X </a>
            </div>
          </li>
        ))}
      </ul>
    );
  }
}
