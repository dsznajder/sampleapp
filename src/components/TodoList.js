import React, { Component } from 'react'

export default class TodoList extends Component {
  removeTodo = todo => () => this.props.removeTodo(todo)

  render() {
    return (
      <ul>
        {this.props.todos.map((todo, index) => (
          <li key={index}>
            {todo.text}
            <a onClick={this.removeTodo(todo)}> X </a>
          </li>
        ))}
      </ul>
    )
  }
}
