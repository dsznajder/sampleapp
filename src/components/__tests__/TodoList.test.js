import React from 'react'
import renderer from 'react-test-renderer'

import TodoList from '../TodoList'

describe('<TodoList />', () => {
  const defaultProps = {
    removeTodo: jest.fn(),
    todos: [],
  }

  const wrapper = renderer.create(<TodoList {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
