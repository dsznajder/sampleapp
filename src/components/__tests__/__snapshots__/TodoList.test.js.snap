// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> render 1`] = `<ul />`;
